app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -218,6 +219,20 @@ sealed class RecentSyncedTabState {
     data class Success(val tabs: List<RecentSyncedTab>) : RecentSyncedTabState()
 }
 
+internal fun RecentSyncedTabState.updateFilteredTabs(blocklistHandler: BlocklistHandler) =
+    if (this is RecentSyncedTabState.Success) {
+        with(blocklistHandler) {
+            val filteredTabs = this@updateFilteredTabs.tabs.filteredByBlocklist()
+            if (filteredTabs.isEmpty()) {
+                RecentSyncedTabState.None
+            } else {
+                RecentSyncedTabState.Success(filteredTabs)
+            }
+        }
+    } else {
+        this
+    }
+
```

Comment:
Would it make sense to move the content of this function into the blocklist handler, such that we had `RecentSyncedTabState.filteredByBlocklist` instead of `List<RecentSyncedTab>.filteredByBlocklist`? 

I think that would allow us to have the call sites mirror the rest of the `filteredByBlocklist` calls more closely.